workflow/workflow_commands.go

```diff
@@ -1445,6 +1451,56 @@ func TraceWorkflow(c *cli.Context) error {
 	return nil
 }
 
+func UpdateWorkflow(c *cli.Context) error {
+	namespace, err := common.RequiredFlag(c, common.FlagNamespace)
+	if err != nil {
+		return err
+	}
+	wid := c.String(common.FlagWorkflowID)
+	rid := c.String(common.FlagRunID)
+	name := c.String(common.FlagName)
+	args, err := common.ProcessJSONInput(c)
+	first_execution_run_id := c.String(common.FlagUpdateFirstExecutionRunID)
+	if err != nil {
+		return err
+	}
+	waitPolicy := c.String(common.FlagUpdateWaitPolicy)
+	if _, ok := updateWaitPolicyMap[waitPolicy]; !ok {
+		return fmt.Errorf("must specify valid wait policy: %v", strings.Join(mapKeysToArray(updateWaitPolicyMap), ", "))
+	}
+	sdk, err := client.GetSDKClient(c)
+	if err != nil {
+		return err
+	}
+	ctx, cancel := common.NewContext(c)
+	defer cancel()
+
+	_, err = sdk.WorkflowService().UpdateWorkflowExecution(ctx, &workflowservice.UpdateWorkflowExecutionRequest{
```

Comment:
I think we may want to use the high level update calls available on the client. Also, we need to know whether the user wants to wait for update accepted by server, accepted by workflow, or completed. Also, if users do want to wait for completed, we need to show the result similar to how we do for queries.

We may need to step back and outline all of the workflow update commands (there will also be ones on getting the result of an already started update I would think).

Reply:
> I think we may want to use the high level update calls available on the client. 
Sounds good.

>  Also, we need to know whether the user wants to wait for update accepted by server, accepted by workflow, or completed. Also, if users do want to wait for completed, we need to show the result similar to how we do for queries.

Good point, I was just about to ask whether we want to process the result in any way, I check the queries and make some updates. 

> We may need to step back and outline all of the workflow update commands (there will also be ones on getting the result of an already started update I would think).

Sounds good to me.

Reply:
Made the changes to have all these. `waitPolicy` and related functions has just been added to sdk client and not yet available. I added the required logic and flags to extend this when these are released. 

Currently either `--name` or `--update-id` should be passed to command, former is to do an update, latter is for getting the result of an update.